spec/solver_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'rspec'
+require_relative '../solver'
+
+RSpec.describe Solver do
+  let(:solver) { Solver.new }
+
+  describe 'factorial and reverse' do
+    context 'Solves for the given methods' do
+      before(:each) do
+        @solver = Solver.new
+      end
+
+      it 'should solve factorial for the given integers' do
+        expect(@solver.factorial(0)).to eq(1)
+        expect(@solver.factorial(1)).to eq(1)
+        expect(@solver.factorial(2)).to eq(2)
+        expect(@solver.factorial(3)).to eq(6)
+        expect(@solver.factorial(4)).to eq(24)
+        expect(@solver.factorial(5)).to eq(120)
+        expect(@solver.factorial(6)).to eq(720)
+      end
```

Comment:
- [ ] As required for the project, kindly ensure that The` #factorial` method has tests that handle the following case:
  * Negative value which raises an exception as an expectation

Reply:
resolved! ✔️